src/main/java/de/teamlapen/vampirism/entity/player/actions/ActionHandler.java

```diff
@@ -348,12 +417,17 @@ public boolean updateActions() {
             ILastingAction<T> action = (ILastingAction<T>) RegUtil.getAction(entry.getKey());
             assert action != null;
             if (newtimer == 0) {
-                action.onDeactivated(player);
-                cooldownTimers.put(entry.getKey(), action.getCooldown(player));
+                deactivateAction(action, true);
+                if(!cooldownTimers.containsKey(entry.getKey())) {
+                    cooldownTimers.put(entry.getKey(), modifiedCooldownTimers.getInt(RegUtil.id(action)));
```

Comment:
If I'm understanding correctly, overriding it here would cause the incorrect timer to be applied. For example, when the action is deactivated manually it needs to have a reduced cooldown based on the remaining duration. This wouldn't work here and would cause it to use the full cooldown. This is because the timer can't be directly removed in `deactivateAction`.